exporter/signalfxexporter/internal/dimensions/dimclient.go

```diff
@@ -319,7 +325,7 @@ func (dc *DimensionClient) makePatchRequest(dim *DimensionUpdate) (*http.Request
 	}
 
 	req, err := http.NewRequestWithContext(
-		context.Background(),
+		ctx,
```

Comment:
This allows the dimension client to shut itself down when requested, instead of relying on an outside caller to handle its lifecycle.